Hadrons/Modules/MGauge/StoutSmearing.hpp

```diff
@@ -109,7 +110,13 @@ void TStoutSmearing<GImpl>::execute(void)
                  << " step" << ((par().steps > 1) ? "s" : "") 
                  << " of stout smearing and rho= " << par().rho << std::endl;
 
-    Smear_Stout<GImpl> smearer(par().rho);
+    int oDim = -1;
+    if(!par().orthogDim.empty())
+    {
+        LOG(Message) << "Only smearing orthogonally to dimension " << par().orthogDim << std::endl;
+	oDim=std::stoi(par().orthogDim);
+    }
+    Smear_Stout<GImpl> smearer(par().rho, oDim);
```

Comment:
orthogDim = -1 calls the constructor smearing with the same constant weight in all 4D directions. Otherwise the smearing is applied in the 3 dimensions orthogonally to orthogDim.